pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -175,6 +187,25 @@ public NamespaceService(PulsarService pulsar) {
         this.bundleSplitListeners = new CopyOnWriteArrayList<>();
         this.localBrokerDataCache = pulsar.getLocalMetadataStore().getMetadataCache(LocalBrokerData.class);
         this.redirectManager = new RedirectManager(pulsar);
+
+        var meter = pulsar.getOpenTelemetry().getMeter();
+        this.lookupRedirectsCounter = meter
+                .counterBuilder("pulsar.broker.lookup.redirect")
+                .setDescription("The number of lookup redirected requests")
+                .build();
+        this.lookupFailuresCounter = meter
+                .counterBuilder("pulsar.broker.lookup.failure")
+                .setDescription("The number of lookup failures")
+                .build();
+        this.lookupAnswersCounter = meter
+                .counterBuilder("pulsar.broker.lookup.answer")
+                .setDescription("The number of lookup responses (i.e. not redirected requests)")
+                .build();
+        this.lookupLatencyHistogram = meter
+                .histogramBuilder("pulsar.broker.lookup.latency")
```

Comment:
IMO, we can only have one attribute `response.status`. The url looks too details to the metrics? if we have 50 or 100 brokers.

pulsar.broker.lookup.request.duration(response.status="connect/redirect/failed")

Reply:
I agree that including the URL provides little value. 

As for the attribute name, we should keep the `pulsar` prefix in place to avoid confusion or clashes with attributes in other namespaces.

I liked @asafm's proposal to split this by a general status (success or failure), as this can further be used by other metrics while also adding details relating to this particular metric where we can (connect/redirect). I think this is an excellent structure to follow going forward.

To reiterate, we'd have a separate histogram record for the following attribute sets (ignoring general resource attributes, such as `pulsar.cluster`):
`pulsar.lookup.response.type=broker,pulsar.response.status=success`
`pulsar.lookup.response.type=redirect,pulsar.response.status=success`
`pulsar.response.status=failure`.

Reply:
@codelipenghui The attribute name is `pulsar.lookup.response.type`, and it can have the following values: `broker` or `redirect`. We are not placing the actual URL as the attribute value.

Given this, are you ok with the 2 attributes suggested and the name?

Name = `pulsar.broker.lookup.request.duration`
Attributes:
1. `pulsar.lookup.response.type`: values can be `broker`, `redirect`
2. `pulsar.response.status`: values can be `success`, `failure`

Reply:
Both `broker, redirect` connote success. Why do we need to have (broker, success), (redirect, success)? For failures, it could be a couple of reason, topic not found or some temporary reason (metadata issue), I'm not sure how the broker can say the reason is belongs to `redirect` or `broker`. So the response type will be absent for the failed lookup response.

https://github.com/apache/pulsar/blob/ccc2ea67c192b4771c9af30c6eaf994fe01958e5/pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java#L974-L982

I think the name defined in Pulsar is little confusing. If the name is LookupResponseCode, it will be easy to comprehend than LookupType :)

IMO, `broker`, `redirect` and `failure` are just like the HTTP response code `200`, `307`, `500`.

Another question about the attribute name. I guess we will have global attributes so we must have `pulsar.lookup` as prefix? Is it possible to have local attribute for `pulsar.lookup.response.type`? It could only work with some metrics related to lookup.

Reply:
Ok, after a good discussion with @codelipenghui, with came up with the following instrument names:

```
pulsar.broker.topic.load.duration
pulsar.broker.topic.load.concurrent.usage
pulsar.broker.topic.load.concurrent.limit

pulsar.broker.request.topic.lookup.duration
pulsar.broker.request.topic.lookup.concurrent.usage
pulsar.broker.request.topic.lookup.concurrent.limit
```

Why?
1. The 3rd name is usually a category. For example, messaging, transaction, offloader, etc. After some discussion, we figured "request" could be a decent category for requests like lookup.
   * The code that uses the lookup semaphore needs to be corrected. The only ones who should be using it are the actual lookup HTTP request and the binary command lookup. The rest are wrong, and they should be removed.
2. Internal operations such as topic load can be implicitly assumed, so we've removed them from the instrument naming.

Regarding the attributes, we were thinking of uniting the two attributes into one:
```
	Attributes:
		pulsar.lookup.response = broker, redirect, failure
```

The reasoning was that, like the HTTP status code, pulsar lookup response (code) can signify different responses. We can revisit that as we advance to other requests in the future - we have marked all of this experimental.

@dragosvictor @codelipenghui @lhotari

Reply:
Name changes sound good. Note that there is no metric `pulsar.broker.topic.load.duration
` as of right now, I'm not sure why it got included in the list.

Reply:
We were brainstorming many metrics :)